### PR TITLE
ARUHA-855 Check authorization on even type publishing

### PR DESCRIPTION
**Pull request by antban**
ARUHA-855

### CODE REVIEW

**Comment by adyach** [comment]
👍

File: src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java [hunk 1]
```diff
@@ -44,6 +44,21 @@ public void validateAuthorization(@Nullable final EventTypeAuthorization auth) t
         }
     }
 
+    public void validateWrite(final EventType eventType) {
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java]: throws ResourceAccessNotAuthorizedException

File: src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java [hunk 2]
```diff
@@ -44,6 +44,21 @@ public void validateAuthorization(@Nullable final EventTypeAuthorization auth) t
...
+        if (eventType.getAuthorization() == null) {
+            return;
+        }
+        final MyEventTypeResource2 resource = new MyEventTypeResource2(
+                eventType.getName(), eventType.getAuthorization());
+        final boolean authorized = authorizationService.isAuthorized(
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/service/AuthorizationValidator.java]: isAuthorized throws PluginException, which could hold some technical information we should not expose to the users.

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

File: src/main/java/org/zalando/nakadi/service/MyEventTypeResource2.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            case ADMIN:
+                return Optional.of(etAuthorization.getAdmins());
+            default:
+                throw new IllegalArgumentException("Operation " + operation + " is not supported");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "AuthorizedResource{event-type='" + name + "}";
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/service/MyEventTypeResource2.java]: @antban closing `'` is missing

**Pull request merged**